query/stdlib/testing/end_to_end_test.go

```diff
@@ -222,91 +222,42 @@ func testFlux(t testing.TB, l *launcher.TestLauncher, file *ast.File) {
 		OrganizationID: l.Org.ID,
 		Compiler:       lang.ASTCompiler{AST: bs},
 	}
+
 	if r, err := l.FluxQueryService().Query(ctx, req); err != nil {
 		t.Fatal(err)
 	} else {
+		results := make( map[string]*bytes.Buffer )
+
 		for r.More() {
 			v := r.Next()
-			if err := v.Tables().Do(func(tbl flux.Table) error {
-				return tbl.Do(func(reader flux.ColReader) error {
-					return nil
-				})
-			}); err != nil {
+
+			if _, ok := results[v.Name()]; !ok {
+				results[v.Name()] = &bytes.Buffer{}
+			}
+			err := execute.FormatResult(results[v.Name()], v)
+			if err != nil {
 				t.Error(err)
 			}
 		}
-	}
-
-	// quirk: our execution engine doesn't guarantee the order of execution for disconnected DAGS
-	// so that our function-with-side effects call to `to` may run _after_ the test instead of before.
-	// running twice makes sure that `to` happens at least once before we run the test.
-	// this time we use a call to `run` so that the assertion error is triggered
-	runCalls := stdlib.TestingRunCalls(pkg)
-	pkg.Files[len(pkg.Files)-1] = runCalls
-
-	bs, err = json.Marshal(pkg)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	req = &query.Request{
-		OrganizationID: l.Org.ID,
-		Compiler:       lang.ASTCompiler{AST: bs},
-	}
-	r, err := l.FluxQueryService().Query(ctx, req)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	needInspect := false
-	for r.More() {
-		v := r.Next()
-		if err := v.Tables().Do(func(tbl flux.Table) error {
-			return tbl.Do(func(reader flux.ColReader) error {
-				return nil
-			})
-		}); err != nil {
-			needInspect = true
+		if err := r.Err(); err != nil {
 			t.Error(err)
 		}
-	}
-	if err := r.Err(); err != nil {
-		t.Error(err)
-		needInspect = true
-	}
-	if needInspect {
-		// Replace the testing.run calls with testing.inspect calls.
-		pkg.Files[len(pkg.Files)-1] = inspectCalls
-		bs, err = json.Marshal(pkg)
-		if err != nil {
-			t.Fatal(err)
-		}
-		req = &query.Request{
-			OrganizationID: l.Org.ID,
-			Compiler:       lang.ASTCompiler{AST: bs},
-		}
-		r, err := l.FluxQueryService().Query(ctx, req)
-		if err != nil {
-			t.Fatal(err)
-		}
-		var out bytes.Buffer
-		defer func() {
-			if t.Failed() {
-				scanner := bufio.NewScanner(&out)
+
+		logFormatted := func( name string, results map[string]*bytes.Buffer ) {
+			if _, ok := results[name]; ok {
+				scanner := bufio.NewScanner(results[name])
 				for scanner.Scan() {
-					t.Log(scanner.Text())
+					t.Log( scanner.Text())
 				}
-			}
-		}()
-		for r.More() {
-			v := r.Next()
-			err := execute.FormatResult(&out, v)
-			if err != nil {
-				t.Error(err)
+			} else {
+				t.Log( "table ", name, " not present in results" )
 			}
 		}
-		if err := r.Err(); err != nil {
-			t.Error(err)
+		if _, ok := results["diff"]; ok {
+			t.Error("diff table was not empty")
+			logFormatted("diff", results)
+			logFormatted("want", results)
+			logFormatted("got", results)
```

Comment:
will this improve the way that test diffs are displayed? At present, they are a little confusing to read.

Reply:
The way it ends up reading is like:

```
diff table was not empty
result "diff"
 <table-data>
result "want"
 <table-data>
result "got"
 <table-data>
```

It won't print any other tables it finds (previously you could see "to1" or other tables made by the test). Also, it will always print in that order. Let me know if there are any other improvements you had in mind!